src/desktop/space/mod.rs

```diff
@@ -32,43 +27,49 @@ mod window;
 
 pub use self::element::*;
 use self::output::*;
-use self::window::*;
 
 use super::WindowSurfaceType;
 
 crate::utils::ids::id_gen!(next_space_id, SPACE_ID, SPACE_IDS);
 
+#[derive(Debug)]
+struct InnerElement<E> {
+    element: E,
+    location: Point<i32, Logical>,
+    outputs: HashSet<Output>,
+}
+
 /// Represents two dimensional plane to map windows and outputs upon.
 #[derive(Debug)]
-pub struct Space {
+pub struct Space<E: SpaceElement> {
     pub(super) id: usize,
     // in z-order, back to front
```

Comment:
Not sure if we still need the elements in back to front z-order. The new renderer abstractions expect them in front to back order and input handling also probably works that way.